src/SmartComponents.AspNetCore/DefaultSmartComponentsBuilder.cs

```diff
@@ -2,21 +2,17 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Microsoft.Extensions.DependencyInjection;
-using SmartComponents.StaticAssets.Inference;
+using Microsoft.Extensions.AI;
 
 namespace Microsoft.AspNetCore.Builder;
 
 internal sealed class DefaultSmartComponentsBuilder(IServiceCollection services) : ISmartComponentsBuilder
 {
-    public ISmartComponentsBuilder WithInferenceBackend<T>(string? name) where T : class, IInferenceBackend
+    public ISmartComponentsBuilder WithInferenceBackend<T>(string? name) where T : class, IChatClient
     {
         if (string.IsNullOrEmpty(name))
         {
-            services.AddSingleton<IInferenceBackend, T>();
-        }
-        else
-        {
-            services.AddKeyedSingleton<IInferenceBackend, T>(name);
+            services.AddSingleton<IChatClient, T>();
```

Comment:
Can you clarify why you're removing support for keyed DI services here?

Reply:
That was an oversight which I can add back. The other thing I just noticed is that the method this is in, WithInferenceBackend, is not referenced anymore. Should we delete it or is there a reason we should keep it?

Reply:
Hey Steve, I added back support for keyed DI services. I was also able to get all the tests working!